commands/previewPush.go

```diff
@@ -231,7 +302,7 @@ DomainLoop:
 }
 
 // InitializeProviders takes (fully processed) configuration and instantiates all providers and returns them.
-func InitializeProviders(cfg *models.DNSConfig, providerConfigs map[string]map[string]string, notifyFlag bool) (notify notifications.Notifier, err error) {
+func InitializeProviders(cfg *models.DNSConfig, providerConfigs map[string]map[string]string, notifyFlag bool) (createdRegistrars map[string]providers.Registrar, createdDnsProviders map[string]providers.DNSServiceProvider, notify notifications.Notifier, err error) {
```

Comment:
This signature has changed a few times in  the last few years.  Each time we add more return values then update all the other code as appropriate.  My suggestion is that we return a pointer to a structure that has fields for all the current return values.  In the future we can add new fields and the signature won't change.

Reply:
Done.